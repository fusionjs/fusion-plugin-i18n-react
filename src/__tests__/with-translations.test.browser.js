// @flow

import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {mount} from 'enzyme';

import {withTranslations} from '../index';

function mockProvider(i18n) {
  class I18NProvider extends Component<*, *> {
    getChildContext() {
      return {i18n};
    }
    render() {
      return React.Children.only(this.props.children);
    }
  }

  I18NProvider.childContextTypes = {
    i18n: PropTypes.object.isRequired,
  };

  return I18NProvider;
}

test('withTranslations() HOC - localeCode', () => {
  const Foo = withTranslations([])(({localeCode, translate}) => {
    return (
      <div>
        {localeCode}
        {translate()}
      </div>
    );
  });

  const Provider = mockProvider({
    localeCode: 'fr_CA',
    translate() {
      return 'foo bar baz';
    },
  });

  expect(
    mount(
      <Provider>
        <Foo />
      </Provider>
    ).html()
  ).toMatchSnapshot();
});
