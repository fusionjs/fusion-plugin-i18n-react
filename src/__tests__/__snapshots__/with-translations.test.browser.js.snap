// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withTranslations() - localeCode 1`] = `"<div>fr_CAfoo bar baz</div>"`;
